src/Simplify.cpp

```diff
@@ -1449,7 +1449,9 @@ class Simplify : public IRMutator {
         } else if (is_one(b)) {
             expr = a;
         } else if (equal(a, b) &&
-                   !is_zero(b)) {
+                   !is_zero(b) &&
+                   // (n/0)/(n/0) should not be simplified to 1
+                   !(div_a && is_zero(div_a->b))) {
```

Comment:
div-by-zero is the only case I'm aware of, though I'm there are probably others. (div-by-Inf and div-by-NaN seem likely as well. Will generalize.)